File: classes/parte-5-funciones-y-modularidad/080-paso-por-referencia/implementaciones/c/main.c
#include <stdio.h>

void doblar(long *p) {
    *p *= 2;
}

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    long antes = n;
    doblar(&n);
    printf("antes=%ld despues=%ld\n", antes, n);
    return 0;
}

File: classes/parte-5-funciones-y-modularidad/080-paso-por-referencia/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func doblar(p *int) {
	*p *= 2
}

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(line))
	antes := n
	doblar(&n)
	fmt.Printf("antes=%d despues=%d\n", antes, n)
}
